hibernate-reactive-core/src/main/java/org/hibernate/reactive/id/insert/ReactiveAbstractReturningDelegate.java

```diff
@@ -98,6 +99,24 @@ private static String createInsert(PreparedStatementDetails insertStatementDetai
 			// Correct  : select id from new table ( insert into LongTypeEntity (id) values (default))
 			return insertStatementDetails.getSqlString().replace( " values ( ))", " (" + identifierColumnName + ") values (default))" );
 		}
+		if( dialect instanceof Oracle12cDialect ) {
+			final String valuesStr = " values ( )";
+			String sql = insertStatementDetails.getSqlString();
+			int index = sql.lastIndexOf( sqlEnd );
+			// remove "returning id" since it's added via
+			if ( index > -1 ) {
+				sql = sql.substring( 0, index );
+			}
+
+			// Oracle is expecting values (default)
+			if ( sql.endsWith( valuesStr ) ) {
+				index = sql.indexOf( valuesStr );
```

Comment:
it's probably better to use `.lastIndexOf` in this case.

Reply:
fixed